src/test/java/org/elasticsearch/cluster/SimpleClusterStateTests.java

```diff
@@ -157,4 +160,51 @@ public void testLargeClusterStatePublishing() throws Exception {
             assertThat(mappingMetadata, equalTo(masterMappingMetaData));
         }
     }
+
+    @Test
+    public void testIndicesOptions() throws Exception {
+        ClusterStateResponse clusterStateResponse = client().admin().cluster().prepareState().clear().setMetaData(true).setIndices("f*")
+                .get();
+        assertThat(clusterStateResponse.getState().metaData().indices().size(), is(2));
+
+        // close one index
+        client().admin().indices().close(Requests.closeIndexRequest("fuu")).get();
+        clusterStateResponse = client().admin().cluster().prepareState().clear().setMetaData(true).setIndices("f*").get();
+        assertThat(clusterStateResponse.getState().metaData().indices().size(), is(1));
+        assertThat(clusterStateResponse.getState().metaData().index("foo").state(), equalTo(IndexMetaData.State.OPEN));
+
+        // expand_wildcards_closed should toggle return only closed index fuu
+        IndicesOptions expandCloseOptions = IndicesOptions.fromOptions(false, true, false, true);
+        clusterStateResponse = client().admin().cluster().prepareState().clear().setMetaData(true).setIndices("f*")
+                .setIndicesOptions(expandCloseOptions).get();
+        assertThat(clusterStateResponse.getState().metaData().indices().size(), is(1));
+        assertThat(clusterStateResponse.getState().metaData().index("fuu").state(), equalTo(IndexMetaData.State.CLOSE));
+
+        // ignore_unavailable set to true should not raise exception on fzzbzz
+        IndicesOptions ignoreUnavailabe = IndicesOptions.fromOptions(true, true, true, false);
+        clusterStateResponse = client().admin().cluster().prepareState().clear().setMetaData(true).setIndices("fzzbzz")
+                .setIndicesOptions(ignoreUnavailabe).get();
+        assertTrue(clusterStateResponse.getState().metaData().indices().isEmpty());
```

Comment:
for consistency: assertThat(clusterStateResponse.getState().metaData().indices().isEmpty(), is(true));